### ISSUE TITLE
optimizer weights not saved into keras.model.save() via TPU training

### ISSUE THREAD

**Issue by SchenbergZY** [tensorflow/tensorflow#27979]
While I need train over 12h on colab TPU, I should train the model, save and reload the model, and continue training. But [optimizer_weights] are not saved into my_model.h5. Here I have my demo code to examine this:

<em>import tensorflow as tf
from tensorflow.keras.layers import Input,Dense
from tensorflow.keras.models import Model
import numpy as np
from keras import backend as K
import os
np.random.seed(1000)
X, y = np.random.rand(200, 50), np.random.randint(2, size=200)
x = Input((50,))
out = Dense(1, activation='sigmoid')(x)
model100 = Model(x, out)
model100.compile(optimizer='adam', loss='binary_crossentropy')
use_tpu=True # @param {type:"boolean"}
if use_tpu:
  assert 'COLAB_TPU_ADDR' in os.environ, 'ERROR: Not connected to a TPU runtime; Maybe you should switch hardware accelerator to TPU for TPU support'
  tpu_address = 'grpc://' + os.environ['COLAB_TPU_ADDR']
  strategy = tf.contrib.tpu.TPUDistributionStrategy(
          tf.contrib.cluster_resolver.TPUClusterResolver(tpu=tpu_address)
  )
  model3 = tf.contrib.tpu.keras_to_tpu_model(model100, strategy=strategy)
model3.fit(X, y, epochs=1,batch_size=8)
model3.save('weights.h5')

import h5py
filename = 'weights.h5'
h555 = h5py.File(filename,'r')
print(list(h555.keys()))</em>

Finally I get the keys as ['model_weights'] but not ['model_weights','optimizer_weights']
I am not sure there are other ways to get optimizer_weights, if yes, please dont hesitate to tell.

**Comment by muddham** [comment]
@SchenbergZY Please confirm us the tensorflow version being used here, is it the default colab tensorflow version 1.13.1. This information helps us to reproduce the issue.

**Comment by jvishnuvardhan** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will open a new issue. Thanks!

**Comment by SchenbergZY** [comment]
> @SchenbergZY Please confirm us the tensorflow version being used here, is it the default colab tensorflow version 1.13.1. This information helps us to reproduce the issue.

Yes it is the default colab.
